### ISSUE TITLE
Background Importer Fails To Find Any Results Even Given CVID

### ISSUE THREAD

**Issue by trumblejoe** [evilhero/mylar#2012]
A few things to mention first-off.

These are series which have already been added to Mylar, named, tagged with metadata. I'm rebuilding an install of Mylar and so this is a fresh import of files which have already been processed and added to my previous Mylar install (so I am certain that these are identifiable by Mylar).

As I am working my way through import results management, if I add series one by one and click Import, it runs without any issues. If I attempt to select and import multiple series at one time running the background importer, that is when it fails. I've been working my way through about 2000 comics right now, and I've done this enough times that it is clear there is something wrong with how the background importer finds comics compared to a single comic series import.

Logs Below.

2018-07-08 19:08:46 | INFO | Update complete
-- | -- | --
2018-07-08 19:08:46 | DEBUG | ReScheduled job: DB Updater to 2018-07-08 17:13:46
2018-07-08 19:08:46 | DEBUG | Refresh sequence set to fire every 5 minutes for 7 day(s)
2018-07-08 19:06:47 | INFO | [IMPORT] Import completed.
2018-07-08 19:06:47 | DEBUG | [IMPORT] I cannot find any results for the given series. I should remove this from the list.
2018-07-08 19:06:47 | DEBUG | [IMPORT] [!none] volume: v1
2018-07-08 19:06:47 | DEBUG | [IMPORT] [!none] dynamicname: xmendrdoom98
2018-07-08 19:06:47 | INFO | [IMPORT][X-Men / Dr. Doom '98 (v1)] Now preparing to import. First I need to determine the highest issue, and possible year(s) of the series.
2018-07-08 19:06:47 | DEBUG | [IMPORT] I cannot find any results for the given series. I should remove this from the list.
2018-07-08 19:06:47 | DEBUG | [IMPORT] [!none] volume: v1
2018-07-08 19:06:47 | DEBUG | [IMPORT] [!none] dynamicname: xmenendangeredspecies
2018-07-08 19:06:47 | INFO | [IMPORT][X-Men: Endangered Species (v1)] Now preparing to import. First I need to determine the highest issue, and possible year(s) of the series.
2018-07-08 19:06:47 | DEBUG | [IMPORT] {'Volume': 'v1', 'ComicName': u"X-Men / Dr. Doom '98", 'DynamicName': u'xmendrdoom98', 'ComicID': None}:
2018-07-08 19:06:47 | DEBUG | [IMPORT] Checking for any valid ComicID's already present within filenames.
2018-07-08 19:06:47 | DEBUG | [IMPORT] {'Volume': 'v1', 'ComicName': u'X-Men: Endangered Species', 'DynamicName': u'xmenendangeredspecies', 'ComicID': None}:
2018-07-08 19:06:47 | DEBUG | [IMPORT] Checking for any valid ComicID's already present within filenames.
2018-07-08 19:06:47 | DEBUG | The following series will now be attempted to be imported: [{'Volume': 'v1', 'ComicName': u'X-Men: Endangered Species', 'DynamicName': u'xmenendangeredspecies', 'ComicID': None}, {'Volume': 'v1', 'ComicName': u"X-Men / Dr. Doom '98", 'DynamicName': u'xmendrdoom98', 'ComicID': None}]
2018-07-08 19:06:47 | INFO | Initiating selected import mode for 2 series.
2018-07-08 19:06:47 | INFO | importing selected series.

**Comment by evilhero** [comment]
According to that log there is no comicid provided in the metadata so it has to rely on file name parsing. If the metadata had the comicid (or you had a cvinfo file in the given directory pointing to the cv url for the series), it would be able to import things without relying on the file parser. 

The mass import / select options use the exact same method to import as selecting an individual option (only difference is the reiteration). The problem herein would lie with the series you're trying to import most likely. 

Importing annuals are tricky, as that annual on CV has no information directly linking it back to the actual series itself, but just to more annuals. The importer won't direcrly attach annuals to a given series if it can't detect the link on CV to which the given annual is linked to. Also  the word annual doesn't directly appear in the name of the issue, which probably doesn't help things either. 

The log doesn't indicate how you have your filenames, so it's difficult to know if the file parser is having problems, or if it cannot find a matching series on CV (which it says it can't, but it should be able to, course CV has changed their api abit which might be affecting the results returned)

**Comment by trumblejoe** [comment]
Updated Mylar when I realized it wasn't current to the most recent commit. After doing so, here is the error message I'm being given from the log. This is only when I use background import, and I know all of these files have CVID's. The ID shows up on the Import Results Management page and these are all files which I have had in a previous install of Mylar. My guess is that the ultimate problem is with your reiteration and that it is losing track of the CVID's.

2018-07-09 06:36:46 | ERROR | Uncaught exception: Traceback (most recent call last):File "/home6/merciless1/Mylar/mylar/logger.py", line 336, in new_runold_run(*args, **kwargs)File "/usr/lib/python2.7/threading.py", line 754, in runself.__target(*self.__args, **self.__kwargs)File "/home6/merciless1/Mylar/mylar/webserve.py", line 4279, in preSearchitmyDB.upsert("searchresults", nVal, cVal)File "/home6/merciless1/Mylar/mylar/db.py", line 170, in upsertself.action(query, valueDict.values() + keyDict.values())File "/home6/merciless1/Mylar/mylar/db.py", line 129, in actionsqlResult = self.connection.execute(query, args)InterfaceError: Error binding parameter 9 - probably unsupported type.
-- | -- | --
2018-07-09 06:36:12 | INFO | [IMPORT][Captain America (v4)] Now preparing to import. First I need to determine the highest issue, and possible year(s) of the series.
2018-07-09 06:36:12 | INFO | Initiating selected import mode for 8 series.
2018-07-09 06:36:12 | INFO | importing selected series.

**Comment by evilhero** [comment]
Enable debug logs and try the import again (I would restart mylar after the error to make sure things are all cleaned out). Unsupported type error would indicate that there's a field that doesn't contain the correct type of data. Can't see the debug lines so it's difficult to see exactly what it's possibly erroring on. 

Again, can't duplicate this on my end at all. Ran against several variations of selections and all import /provide options after import without errors.

Can you provide your OS, python version and the branch /commit of mylar you're running - as these are all important parts to the overall puzzle as well.

Aside from that, providing the mylar.db file would allow me to load up the exact import sequence as you have it, and see for myself exactly what's happening.

**Comment by trumblejoe** [comment]
Configuration OptionsMylar Version: development-- git build 22de99943bc63039e33d27c3333ca490f2a4b8fc.
Python Version : 2.7.13Language : en_US.UTF-8
--
Linux lw701 4.9.0-6-amd64 #1 SMP Debian 4.9.88-1+deb9u1 (2018-05-07) x86_64 GNU/Linux

Log attached. Database attached. 

For the series which have failed, they all look like the following : 


X-Men: Legacy (v1)  [20691] | 2010 | 207 | No Results | 2018-07-09 | [Remove] [Select]

However, Select does nothing except reload the page.

------- (As soon as I said that, I had an obvious idea) -----------
I turned on the Web Developer console and reloaded the page, it listed an uncaught TypeError. Please see the javascript log below.

This is as far as I can go tonight, I'm off to sleep, 2am here.

[MylarDebug.zip](https://github.com/evilhero/mylar/files/2174788/MylarDebug.zip)

**Comment by evilhero** [comment]
Welp, that was interesting. If you happened to select multiple series to import that had a comicid listed, it wasn't being passed properly. However if you chose the Import option from the Options column it would pass the ComicID properly, as would if you performed a Mass Import option via the button at the  top of the page. Suffice to say, not quite sure when this changed so that it didn't work, but I have it working in my local repo currently.

The other thing was that if the status was No Results - the option to 'Select' something should not have presented itself, as there are no results to display (thus why when you clicked on Select, it reloaded the page). Fixed this too so now if it shows No Results, the Select option is not presented (it will only appear when the status is Manual Intervention).

In looking at this more, if a result is returned as No Results there should be some way to manually specify the ComicID to which it should belong to, or at the very least view the search results so that one can be manually picked (slightly harder but just because of the way the importer stores the data). I'm going to see about including both options, but the fix for the ComicID not being used, and the No Results/Select bug will be in the next development commit up shortly.

As far as the javascript, I'll have to delve into that abit more as it's definitely not a strong point - it looks like it's having a problem with the datatable, albeit it's not stopping it from displaying the table, so it's probably a minor error of some kind.

Cheers for passing me the mylar.db file too - that was exactly what I needed to see the error occurring.
